src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -399,58 +415,35 @@ private static boolean isSingleLineIf(DetailAST literalIf) {
      */
     private static boolean isSingleLineLambda(DetailAST lambda) {
         final DetailAST block = lambda.getLastChild();
-        return lambda.getLineNo() == block.getLineNo();
+        return isOnSameLine(lambda, block);
     }
 
     /**
-     * Checks if current case statement is single-line statement, e.g.:
+     * Checks if switch member (case or default statement) is single-line statement, e.g.:
      * <p>
      * {@code
      * case 1: doSomeStuff(); break;
      * case 2: doSomeStuff(); break;
      * case 3: ;
+     * default: doSomeStuff();break;
      * }
      * </p>
-     * @param literalCase {@link TokenTypes#LITERAL_CASE case statement}.
-     * @return true if current case statement is single-line statement.
+     * @param ast {@link TokenTypes#LITERAL_CASE case statement} or
+     * {@link TokenTypes#LITERAL_DEFAULT default statement}.
+     * @return true if current switch member is single-line statement.
      */
-    private static boolean isSingleLineCase(DetailAST literalCase) {
+    private static boolean isSingleLineSwitchMember(DetailAST ast) {
         boolean result = false;
-        final DetailAST slist = literalCase.getNextSibling();
+        final DetailAST slist = ast.getNextSibling();
         if (slist == null) {
             result = true;
         }
         else {
             final DetailAST caseBreak = slist.findFirstToken(TokenTypes.LITERAL_BREAK);
             if (caseBreak != null) {
```

Comment:
I have doubts about this code. It means "a single-line `case`/`default` must end with a `break`, otherwise it needs a pair of braces".
For example, this code raises two violations regardless of `allowSingleLineStatement` option:
```
    switch (0) {
      case 1: return; // false violation
      default: throw new RuntimeException(""); // false violation
    }
```
and they can not be fixed by adding a trailing `break`. Why not check the last child's line? Something like
```
    private static boolean isSingleLineSwitchMember(DetailAST ast) {
        final DetailAST slist = ast.getNextSibling();
        return slist == null || slist.getLastChild() == null
                || isOnSameLine(ast, slist.getLastChild());
    }
```

Reply:
Good catch, thank you.

I didn't look deep into that method, it was already there before. Looks like we found and fixed another bug :)

Corrected, also I added test cases.

Reply:
Thank you. Please update the regression report.